### PR TITLE
[sync] release one rlock and change default value at getMaxPeerHeight

### PR DESCRIPTION
**Pull request by JackyWYX**
## Issue

Release one rlock when iterating peers in sync. And change the default return result of getMaxPeerHeight as max uint64

## Test

Passed local tests.

### CODE REVIEW

File: api/service/legacysync/syncing.go [hunk 1]
```diff
@@ -1029,7 +1033,7 @@ func (ss *StateSync) RegisterNodeInfo() int {
 
 // getMaxPeerHeight gets the maximum blockchain heights from peers
 func (ss *StateSync) getMaxPeerHeight(isBeacon bool) uint64 {
-	maxHeight := uint64(0)
+	maxHeight := uint64(math.MaxUint64)
```
Comment (rlan35) [inline api/service/legacysync/syncing.go:1036]: actually, seems the change won't work:

if response != nil && maxHeight < response.BlockHeight {
				maxHeight = response.BlockHeight
			}

Comment (JackyWYX) [reply api/service/legacysync/syncing.go:1036]: Good point. Fixed and thanks. Please review again.

**Comment by rlan35** [comment]
please fix travis.

**Review by rlan35 — APPROVED** [review]

**Comment by rlan35** [comment]
This PR was causing "Error: SyncStatus.TargetIndex is negative: /network/status: /network/status not attempting retry: unable to initialize asserter" on rosetta tests.

**Comment by JackyWYX** [comment]
Noted. Will come back to this later

**Comment by sophoah** [comment]
This PR has been tested on a few multiple RPC nodes and was confirmed to fix the node-sync status so far. With travis build fix, I'll merge that one now

**Pull request merged**